oddt/surface.py

```diff
@@ -0,0 +1,142 @@
+                             'molecule.protein property must be set to True')
+        no_hoh = atom_dict['resname'] != 'HOH'
+        atom_dict = atom_dict[no_hoh]
+
+    # Take a molecule's coordinates and atom radii and scale if necessary
+    coords = atom_dict['coords'] * scaling
+    radii = atom_dict['radius'] * scaling
+
+    # Create a ball for each atom in the molecule
+    ball_dict = {radius: ball(radius, dtype=bool) for radius in set(radii)}
+    ball_radii = np.array([ball_dict[radius].shape[0] for radius in radii])
+
+    # Transform the coordinates because the grid starts at (0, 0 ,0)
+    min_coords = np.min(coords, axis=0)
+    max_rad = np.max(ball_radii, axis=0)
+    adjusted = np.round(coords - min_coords + max_rad * 5).astype(np.int64)
+    offset = adjusted[0] - coords[0]
+
+    # Calculate boundries in the grid for each ball.
+    ball_coord_min = (adjusted.T - np.floor(ball_radii / 2).astype(np.int64)).T
+    ball_coord_max = (ball_coord_min.T + ball_radii).T
+
+    # Create the grid
+    grid = np.zeros(shape=ball_coord_max.max(axis=0) + int(8 * scaling), dtype=bool)
+    # Place balls in grid
+    for radius, coord_min, coord_max in zip(radii, ball_coord_min, ball_coord_max):
+        grid[coord_min[0]:coord_max[0],
+             coord_min[1]:coord_max[1],
+             coord_min[2]:coord_max[2]] += ball_dict[radius]
+    spacing = (1 / scaling,) * 3
+
+    # Hole-filling with morphological closing
+    grid = binary_closing(grid, ball(probe_radius * 2 * scaling))
+
+    # Marching cubes
+    verts, faces, _, _ = measure.marching_cubes_lewiner(grid, level=0, step_size=1, spacing=spacing)
+    # Verts already scaled by the marching cubes function (spacing parameter)
+    # Only need to scale the offset
+    return verts - offset / scaling, faces
+
+
+def find_surface_residues(molecule, max_dist=None, scaling=1):
+    """Finds residues close to the molecular surface using
+    generate_surface_marching_cubes.
+
+    Parameters
+    ----------
+    molecule : oddt.toolkit.Molecule
+        Molecule to find surface residues in.
+
+    max_dist : array_like, numeric or None (default = None)
+        Maximum distance from the surface where residues would
+        still be considered close. If None, compares distances
+        to radii of respective atoms.
+
+    scaling : float (default = 1)
+        Expands the grid in which computation is done by
+        generate_surface_marching_cubes by a factor of scaling.
+        Results in a more accurate representation of the surface,
+        and therefore more accurate computation of distances
+        but increases computation time.
+
+    Returns
+    -------
+    atom_dict : numpy array
+        An atom_dict containing only the surface residues
+        from the original molecule.
+    """
+    # Copy the atom_dict, remove waters
+    atom_dict = molecule.atom_dict
+    mask = (atom_dict['resname'] != 'HOH') & (atom_dict['atomicnum'] != 1)
```

Comment:
There is no hint in docstrings this (HOH removal) happen. We should at least mention it.